uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-tab-filters.component.js

```diff
@@ -33,35 +35,51 @@ define([
      */
     vm.$onChanges = function (changes) {
       if (changes.absencePeriods && vm.absencePeriods.length) {
-        vm.filters.absencePeriod = getCurrentAbsencePeriod();
+        vm.filters.absence_period = getCurrentAbsencePeriod().id;
       }
 
       if (changes.absenceTypes && vm.absenceTypes.length) {
-        vm.filters.absenceType = getFirstAbsenceTypeByTitle();
+        vm.filters.absence_type = getFirstAbsenceTypeByTitle().id;
+      }
+
+      if (changes.loggedInContactId) {
+        vm.filters.managed_by = vm.loggedInContactId;
       }
 
-      if (vm.filters.absencePeriod && vm.filters.absenceType) {
+      if (filtersAreReady()) {
         emitOnFiltersChange();
       }
     };
 
     /**
      * This function is used on the view when the user clicks on *filter* button.
      */
-    vm.filter = function () {
+    vm.doFilter = function () {
       emitOnFiltersChange();
```

Comment:
because $scope.$emit('leaveBalanceFiltersUpdated', vm.filters); can be called in two different places. If I need to change the event name or the parameters I just change `emitOnFiltersChange`. Should I remove `emitOnFiltersChange` ?